v2/go.mod

```diff
@@ -1,11 +1,12 @@
 module github.com/bugsnag/bugsnag-go/v2
 
-go 1.15
+go 1.18
```

Comment:
This is needed for generics. If this upgrade is problematic, I can remove generics from AsyncRunner.